src/shared/components/mutationTable/MutationTable.tsx

```diff
@@ -1089,4 +1125,33 @@ export default class MutationTable<
             />
         );
     }
+
+    protected getMutations() {
+        let data: Mutation[][] | undefined = [];
+        if (this.props.dataStore) {
+            data = this.props.dataStore.allData;
+        } else if (this.props.data) {
+            data = this.props.data;
+        }
+        return data;
+    }
+
+    /**
+     * searches all available mutation data for any mutation with the passed field key
+     * @param property the field name to search for
+     * @returns true if any available mutation is annotated with the passed property name
+     */
+    protected anyMutationHasASCNProperty(property: string): boolean {
```

Comment:
this could potentially be a bottleneck if it get called multiple times with same property.  you could put memoize decorator on it and then should solve the problem.

Reply:
It doesn't feel right to tie this logic to the `MutationTable` component. If possible, this should be handled somewhere upstream (possibly in one of the stores) and stored as a map, something like `[{property: string}: booelan]` considering that all the properties are defined as constants.